src/script/scriptBuilder.zig

```diff
@@ -0,0 +1,239 @@
+        } else if (data.len >= Opcode.OP_PUSHDATA1.toBytes()) {
+            if (data.len > 75 and data.len <= 255) {
+                _ = try self.script.append(76);
+                const size = &[1]u8{@intCast(data.len)};
+                const n = std.mem.readInt(u8, size, .little);
+                _ = try self.script.append(n);
+
+                for (data) |byte| {
+                    try self.script.append(byte);
+                }
+            } else if (data.len >= 256 and data.len <= 65535) {
+                _ = try self.script.append(77);
+                const length: u16 = @intCast(data.len);
+                const n: [2]u8 = [2]u8{
+                    @intCast(length & 0xFF), // lower byte (least significant byte)
+                    @intCast((length >> 8) & 0xFF), // upper byte (most significant byte)
+                };
+                _ = try self.script.append(n[0]);
+                _ = try self.script.append(n[1]);
+
+                for (data) |byte| {
+                    try self.script.append(byte);
+                }
+            } else if (data.len > 65535 and data.len <= 4294967295) {
+                _ = try self.script.append(78);
+                const length: u32 = @intCast(data.len);
+                const n: [4]u8 = [4]u8{
+                    @intCast(length & 0xFF), // lower byte (least significant byte)
+                    @intCast(length >> 8 & 0xFF), // next byte
+                    @intCast(length >> 16 & 0xFF), // next byte
+                    @intCast((length >> 24) & 0xFF), // upper byte (most significant byte)
+                };
+                _ = try self.script.append(n[0]);
+                _ = try self.script.append(n[1]);
+                _ = try self.script.append(n[2]);
+                _ = try self.script.append(n[3]);
+
+                for (data) |byte| {
+                    try self.script.append(byte);
+                }
+            }
+        } else {
+            _ = try self.script.append(Opcode.OP_PUSHDATA1.toBytes());
+        }
+        return self;
+    }
+
+    // Deallocate all resources used by the Engine
+    pub fn deinit(self: *ScriptBuilder) void {
+        self.script.deinit();
+    }
+
+    ///Execute the script. It creates and returns an engine instance.
+    ///
+    /// Returns
+    /// Instance of the engine
+    pub fn build(self: *ScriptBuilder) !engine.Engine {
+        const script = Script.init(self.script.items);
+
+        const script_engine = engine.Engine.init(self.allocator, script, .{});
+        return script_engine;
+    }
+};
+
+test "ScriptBuilder Smoke test" {
+    var sb = try ScriptBuilder.new(std.testing.allocator, 3);
+    defer sb.deinit();
+
+    var e = try (try (try (try sb.addInt(1)).addInt(2)).addOpcode(Opcode.OP_ADD)).build();
+
+    try e.execute();
+
+    defer e.deinit();
+
+    try std.testing.expectEqual(1, e.stack.len());
+
+    const element0 = try e.stack.popInt();
+
+    try std.testing.expectEqual(3, element0);
+}
```

Comment:
https://github.com/zig-bitcoin/btczee/pull/113#discussion_r1774829228

You are executing the script. I explained you in this message why you shouldn't and what you should test instead.